include/common/enum_errcode.h

```diff
@@ -57,20 +57,45 @@ static inline constexpr auto WasmPhaseStr = []() constexpr {
     (defined(_MSVC_LANG) && _MSVC_LANG > 201402L)
 namespace WasmEdge {
 
+/// Error category C++ enumeration class.
+enum class ErrCategory : uint8_t {
+#define UseErrCategory
+#define Line(NAME, VALUE) NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseErrCategory
+};
+
+} // namespace WasmEdge
+#endif
+
+/// Error category C enumeration.
+enum WasmEdge_ErrCategory {
+#define UseErrCategory
+#define Line(NAME, VALUE) WasmEdge_ErrCategory_##NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseErrCategory
+};
+
+#if (defined(__cplusplus) && __cplusplus > 201402L) ||                         \
+    (defined(_MSVC_LANG) && _MSVC_LANG > 201402L)
+namespace WasmEdge {
+
 /// Error code C++ enumeration class.
-enum class ErrCode : uint8_t {
+enum class ErrEnum : uint32_t {
```

Comment:
What's the difference between `ErrEnum` and `ErrCategory`?

Reply:
`ErrEnum` is the origin `ErrCode` to specify the WASM runtime error codes.
`ErrCategory` is the error code category. Currently `WASM` is the `ErrEnum` collections, and `UserErrorLevel` is the user-defined error codes. In the future, we can extend the `WasiNN` or the `WasiCrypto` into the category to separate the error codes from `ErrEnum`. It means, the category size can be extended into 256 (8-bit) in maximum.